packages/trader/src/AppV2/Hooks/useActiveSymbols.ts

```diff
@@ -11,108 +10,117 @@ import {
+import { ActiveSymbols, ActiveSymbolsResponse } from '@deriv/api-types';
 import { useTraderStore } from 'Stores/useTraderStores';
-import { ContractType } from 'Stores/Modules/Trading/Helpers/contract-type';
+import useContractsForCompany from './useContractsForCompany';
+import { useDtraderQuery } from './useDtraderQuery';
 
 const useActiveSymbols = () => {
-    const [activeSymbols, setActiveSymbols] = useState<ActiveSymbols | []>([]);
     const { client, common } = useStore();
-    const { is_logged_in } = client;
+    const { loginid, is_switching } = client;
     const { showError } = common;
     const {
         active_symbols: symbols_from_store,
+        processContractsForV2,
         contract_type,
-        has_symbols_for_v2,
         is_vanilla,
         is_turbos,
         onChange,
         setActiveSymbolsV2,
         symbol,
-        is_trade_component_mounted,
     } = useTraderStore();
+    const [activeSymbols, setActiveSymbols] = useState<ActiveSymbols | []>(symbols_from_store);
+
+    const { available_contract_types, is_fetching_ref: is_contracts_loading_ref } = useContractsForCompany();
 
     const default_symbol_ref = useRef('');
-    const previous_logged_in = usePrevious(is_logged_in);
-    const previous_contract_type = usePrevious(contract_type);
+
+    const trade_types_with_barrier_category = [
+        TRADE_TYPES.RISE_FALL,
+        TRADE_TYPES.RISE_FALL_EQUAL,
+        TRADE_TYPES.HIGH_LOW,
+    ] as string[];
 
     const getContractTypesList = () => {
         if (is_turbos) return [CONTRACT_TYPES.TURBOS.LONG, CONTRACT_TYPES.TURBOS.SHORT];
         if (is_vanilla) return [CONTRACT_TYPES.VANILLA.CALL, CONTRACT_TYPES.VANILLA.PUT];
         return getContractTypesConfig()[contract_type]?.trade_types ?? [];
     };
 
-    const fetchActiveSymbols = useCallback(
-        async () => {
-            let response;
-
-            const trade_types_with_barrier_category = [
-                TRADE_TYPES.RISE_FALL,
-                TRADE_TYPES.RISE_FALL_EQUAL,
-                TRADE_TYPES.HIGH_LOW,
-            ] as string[];
-            const barrier_category = ContractType.getBarrierCategory(contract_type).barrier_category;
-
-            const request = {
-                active_symbols: 'brief',
-                contract_type: getContractTypesList(),
-                ...(trade_types_with_barrier_category.includes(contract_type) && barrier_category
-                    ? { barrier_category: [barrier_category] }
-                    : {}),
+    const { barrier_category } = (available_contract_types?.[contract_type]?.config || {}) as any;
+
+    const isQueryEnabled = useCallback(() => {
+        if (!available_contract_types || is_contracts_loading_ref.current || is_switching) return false;
+        return true;
+    }, [available_contract_types, is_switching, is_contracts_loading_ref]);
+
+    const getContractType = () => {
+        if (isTurbosContract(contract_type)) {
```

Comment:
O think we can reuse `is_vanilla` and `is_turbos` from store and get rid of `isTurbosContract`/`isVanillaContract`. Please, ignore if irrelevant

Reply:
the idea is not to use many things from the store